include/itertools.hpp

```diff
@@ -13,6 +13,42 @@ static inline void report() {
 	puts(__PRETTY_FUNCTION__);
 }
 
+template<typename T>
+class EnumSeqIterator {
+	T _value;
+
+public:
+	explicit EnumSeqIterator(T value) : _value(value) {}
+
+	EnumSeqIterator &operator++() {
+		_value = (T)(_value + 1);
+		return *this;
+	}
+
+	auto operator*() const { return _value; }
+
+	friend auto operator==(EnumSeqIterator const &lhs, EnumSeqIterator const &rhs) {
+		return lhs._value == rhs._value;
+	}
+
+	friend auto operator!=(EnumSeqIterator const &lhs, EnumSeqIterator const &rhs) {
+		return lhs._value != rhs._value;
+	}
+};
+
+template<typename T>
+class EnumSeq {
+	T _start;
+	T _stop;
+
+public:
+	explicit EnumSeq(T stop) : _start((T)0), _stop(stop) {}
```

Comment:
```suggestion
	explicit EnumSeq(T stop) : _start((T)0), _stop(stop) {}
```

Given the lack of special syntax, I think I find it more readable if both bounds are always required. (And it's just three extra characters.)

Reply:
We'd have to say `EnumSeq((enum Foo)0, FOO_INVALID)`, not just `EnumSeq(0, FOO_INVALID)`. Or just as bad, `EnumSeq(FOO_BAR, FOO_INVALID)` and rely on `FOO_BAR` to be first.

Reply:
Actually, I think it should be possible to use [template deduction guides](https://en.cppreference.com/w/cpp/language/class_template_argument_deduction#User-defined_deduction_guides) to have the type be deduced from the second argument... but that's probably too involved for what it brings.